app/src/main/java/org/glucosio/android/db/Migration.java

```diff
@@ -20,126 +20,120 @@
-         String preferred_unit_a1c;
-         String preferred_unit_weight;
-         String preferred_range;
-         int custom_range_min;
-         int custom_range_max;
-
-         class Reminder
-         @PrimaryKey long id;
-         Date alarmTime;
-         boolean oneTime;
-         String metric;
-
-         class CholesterolReading
-         @PrimaryKey long id;
-         int totalReading;
-         int LDLReading;
-         int HDLReading;
-         Date created;
-
-         class GlucoseReading
-         @PrimaryKey long id;
-         int reading;
-         String reading_type;
-         String notes;
-         int user_id;
-         Date created;
-
-         class KetoneReading
-         @PrimaryKey long id;
-         double reading;
-         Date created;
-
-         class PressureReading
-         @PrimaryKey long id;
-         int minReading;
-         int maxReading;
-         Date created;
-
-         class WeightReading
-         @PrimaryKey long id;
-         int reading;
-         Date created;
-
-         class HB1ACReading
-         @PrimaryKey long id;
-         double reading;
-         Date created;
-         ************************************************/
-
         if (oldVersion == 0) {
-            schema.create("WeightReading")
-                    .addField("id", Long.class, FieldAttribute.PRIMARY_KEY, FieldAttribute.REQUIRED)
-                    .addField("created", Date.class)
-                    .addField("reading", Integer.class, FieldAttribute.REQUIRED);
-
-            schema.create("PressureReading")
-                    .addField("id", Long.class, FieldAttribute.PRIMARY_KEY, FieldAttribute.REQUIRED)
-                    .addField("created", Date.class)
-                    .addField("minReading", Integer.class, FieldAttribute.REQUIRED)
-                    .addField("maxReading", Integer.class, FieldAttribute.REQUIRED);
-
-            schema.create("KetoneReading")
-                    .addField("id", Long.class, FieldAttribute.PRIMARY_KEY, FieldAttribute.REQUIRED)
-                    .addField("created", Date.class)
-                    .addField("reading", Double.class, FieldAttribute.REQUIRED);
-
-            schema.create("HB1ACReading")
-                    .addField("id", Long.class, FieldAttribute.PRIMARY_KEY, FieldAttribute.REQUIRED)
-                    .addField("created", Date.class)
-                    .addField("reading", Integer.class, FieldAttribute.REQUIRED);
-
-            schema.create("CholesterolReading")
-                    .addField("id", Long.class, FieldAttribute.PRIMARY_KEY, FieldAttribute.REQUIRED)
-                    .addField("created", Date.class)
-                    .addField("totalReading", Integer.class, FieldAttribute.REQUIRED)
-                    .addField("LDLReading", Integer.class, FieldAttribute.REQUIRED)
-                    .addField("HDLReading", Integer.class, FieldAttribute.REQUIRED);
+            schema.create(RObject.WEIGHT.key())
```

Comment:
Could extract a method for each migration, making easy to read the code and avoiding a massive function body.

Reply:
Extracted